oak-run/src/main/java/org/apache/jackrabbit/oak/run/DataStoreCommand.java

```diff
@@ -462,11 +472,36 @@ private void binaryProperties(NodeState state, String path, ReferenceCollector c
 
         private void traverseChildren(NodeState state, String path, ReferenceCollector collector) {
             binaryProperties(state, path, collector);
-            for (ChildNodeEntry c : state.getChildNodeEntries()) {
+            for (ChildNodeEntry c : getChildNodeEntries(state)) {
                 traverseChildren(c.getNodeState(), PathUtils.concat(path, c.getName()), collector);
             }
         }
 
+        private Iterable<? extends ChildNodeEntry> getChildNodeEntries(NodeState state) {
+            if (useDirListing) {
+                PropertyState dirListing = state.getProperty(PROP_DIRLISTING);
```

Comment:
I wonder if this property "dirListing" is already set? Or is this something we need to add?

Reply:
This is already in use by oak-lucene: https://github.com/apache/jackrabbit-oak/blob/jackrabbit-oak-1.38.0/oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/directory/OakDirectory.java#L68 and is the reason to introduce it. See also the description in OAK-9435.

Reply:
Hmm, I wasn't aware the field is public. I will update the PR and use the existing constant.